PhysicsTools/PatAlgos/plugins/PATMuonProducer.cc

```diff
@@ -201,16 +201,16 @@ PATMuonProducer::~PATMuonProducer()
 {
 }
 
-std::unique_ptr<GlobalPoint> PATMuonProducer::getMuonDirection(const reco::MuonChamberMatch& chamberMatch,
-							       const edm::ESHandle<GlobalTrackingGeometry>& geometry,
-							       const DetId& chamberId)
+std::optional<GlobalPoint> PATMuonProducer::getMuonDirection(const reco::MuonChamberMatch& chamberMatch,
```

Comment:
was this a fix to a clang warning or just an exercise in C++17?

Reply:
it looks like it is both

Reply:
It was both. The compiler warning was related to how the return value of the function was being used. I originally only fixed that, but then I realized `std::optional` is a much better fit for what the code was trying to do.